lib/action_client/callbacks/after_submit.rb

```diff
@@ -0,0 +1,60 @@
+module ActionClient
+  module Callbacks
+    class AfterSubmit
+      def initialize(with_status: nil, &block)
+        @with_status = with_status || (100..599)
+        @block = block
+      end
+
+      def call(status, headers, body)
+        http_statuses = Array(with_status).map do |status_code|
+          Rack::Utils.status_code(status_code)
+        end
+
+        if http_statuses.any? { |code| Array(code).include?(status) }
```

Comment:
Similar comment to #5: can this be `http_statuses.include?(status)`?

Also, given that we have this pattern showing up in two places, is there value in extracting a class? e.g. `HttpStatusFilter.new(with_status).include?(status)`